src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -44,6 +47,25 @@
    (every? keyword? (keys service-graph))
    (every? (some-fn ifn? service-graph?) (vals service-graph))))
 
+(defn inc-restart-counter!
+  "Increments the counter in a restart-file for purposes of supporting HUP behavior"
+  [restart-file-path]
+  (try
+    (if (fs/exists? restart-file-path)
+      (if (and (fs/writeable? restart-file-path) (fs/readable? restart-file-path))
+        (spit restart-file-path (inc (Long/parseLong (slurp restart-file-path))))
+        (log/error "Restart File is not readable and/or writeable"))
+      (let [dir-path (fs/parent restart-file-path)]
+        (fs/mkdirs dir-path)
+        (spit restart-file-path "0")
+        (inc-restart-counter! restart-file-path)))
+    (catch ArithmeticException e
+      (spit restart-file-path "0")
+      (log/warn "Number of restarts has exceeded Long/MAX_VALUE, resetting file to 0"))
```

Comment:
hmmmmmmmmmmmmmmmmmm i was thinking about this as well and could not decide. would it be bad if we skipped 0? jumping to 1 might be intuitive since it means a service has started once. i feel iffy about skipping 0.. what is your reasoning on this?